File: src/main/java/org/stevenlowes/tools/lifx_controller/Messages/Device/EchoResponse.java
package org.stevenlowes.tools.lifx_controller.Messages.Device;

import org.stevenlowes.tools.lifx_controller.Messages.DataTypes.Payload;

public class EchoResponse extends Payload {
    private final int code = 59;
    private byte[] payload;            // 64-Bytes

    public EchoResponse() {
        payload = new byte[64];
    }

    public EchoResponse(byte[] payload) {
        this.payload = payload;
    }

    public EchoResponse(EchoResponse echoResponse) {
        payload = echoResponse.payload;
    }

    public int getCode() {
        return code;
    }

    public byte[] getPayload() {
        return payload;
    }

    public void setPayload(byte[] payload) {
        this.payload = payload;
    }

    public void setFromCommandByteArray(byte[] byteArray) {
        for (int i = 99; i > 35; i--) {
            payload[(-1 * i) + 99] = byteArray[i];
        }
    }

}

File: src/main/java/org/stevenlowes/tools/lifxcontroller/messages/device/EchoResponse.kt
package org.stevenlowes.tools.lifxcontroller.messages.device

import org.stevenlowes.tools.lifxcontroller.commander.CommonMethods
import org.stevenlowes.tools.lifxcontroller.messages.datatypes.SetOnlyPayload

class EchoResponse(val bytes: ByteArray? = CommonMethods.randomBytes(64)) : SetOnlyPayload(59) {
    constructor(echoResponse: EchoResponse): this(echoResponse.bytes)

    override fun setFromCommandByteArray(byteArray: ByteArray) {
        for (i in 99 downTo 36) {
            //TODO why do we need this non-null ...
            bytes!![-1 * i + 99] = byteArray[i]
        }
    }

}
